platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -0,0 +1,181 @@
+        final Builder newRosterState =
+                previousRosterState.copyBuilder().candidateRosterHash(incomingCandidateRosterHash);
+        removeRoster(previousCandidateRosterHash);
+        storeRoster(candidateRoster, incomingCandidateRosterHash, newRosterState);
+    }
+
+    /**
+     * Sets the Active roster.
+     * This will be called to store a new Active Roster in the state.
+     * The roster must be valid according to rules codified in {@link com.swirlds.platform.roster.RosterValidator}.
+     *
+     * @param roster an active roster to set
+     * @param round the round number in which the roster became active.
+     *              It must be a positive number greater than the round number of the current active roster.
+     */
+    public void putActiveRoster(@NonNull final Roster roster, final long round) {
+        requireNonNull(roster);
+        RosterValidator.validate(roster);
+
+        // update the roster state
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final List<RoundRosterPair> roundRosterPairs = new LinkedList<>(previousRosterState.roundRosterPairs());
+        if (!roundRosterPairs.isEmpty()) {
+            final RoundRosterPair activeRosterPair = roundRosterPairs.getFirst();
+            if (round < 0 || round <= activeRosterPair.roundNumber()) {
+                throw new IllegalArgumentException(
+                        "incoming round number must be greater than the round number of the current active roster.");
+            }
+        }
+        final Bytes activeRosterHash = RosterUtils.hash(roster).getBytes();
+        roundRosterPairs.addFirst(new RoundRosterPair(round, activeRosterHash));
+
+        if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+            final RoundRosterPair lastRemovedRoster = roundRosterPairs.removeLast();
+            removeRoster(lastRemovedRoster.activeRosterHash());
+
+            // At this phase of the implementation, the roster state has a fixed size limit for active rosters.
+            // Future implementations (e.g. DAB) can modify this.
+            if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+                // additional safety check to ensure that the roster state does not contain more than set limit.
+                throw new IllegalStateException(
+                        "Active rosters in the Roster state cannot be more than  " + MAXIMUM_ROSTER_HISTORY_SIZE);
+            }
+        }
+
+        final Builder newRosterState = previousRosterState
+                .copyBuilder()
+                .candidateRosterHash(Bytes.EMPTY)
+                .roundRosterPairs(roundRosterPairs);
+        // since a new active roster is being set, the existing candidate roster is no longer valid
+        // so we remove it if it meets removal criteria.
+        removeRoster(previousRosterState.candidateRosterHash());
```

Comment:
This may or may not be true depending on the business logic that calls these store methods. I'm unsure if this particular piece of logic belongs directly to this method.